spring-context/src/main/java/org/springframework/cache/interceptor/CacheAspectSupport.java

```diff
@@ -1136,12 +1139,30 @@ public Object findInCaches(CacheOperationContext context, Cache cache, Object ke
 				if (adapter.isMultiValue()) {
 					return adapter.fromPublisher(Flux.from(Mono.fromFuture(cachedFuture))
 							.switchIfEmpty(Flux.defer(() -> (Flux) evaluate(null, invoker, method, contexts)))
-							.flatMap(v -> evaluate(valueToFlux(v, contexts), invoker, method, contexts)));
+							.flatMap(v -> evaluate(valueToFlux(v, contexts), invoker, method, contexts))
+							.onErrorResume(RuntimeException.class, ex -> {
+								try {
+									getErrorHandler().handleCacheGetError((RuntimeException) ex, cache, key);
+									return Flux.defer(() -> (Flux) evaluate(null, invoker, method, contexts));
```

Comment:
I'm not entirely convinced that the `defer` is necessary here (similarly to how there's no defer in the `flatMap` above). was there a particular reason for it?

Reply:
You right it's not necessary here.
Removed it